docs/pages/guides/middleware.mdx

```diff
@@ -50,4 +50,26 @@ const frames = createFrames({
     }),
   ],
 });
+```
+
+## Writing your own middleware
+
+You should make sure to clone any request you use, in order to not break other middleware. For example you can only call `const body = await ctx.request.clone().json();` once on a request.
+
```

Comment:
can we not just pass a clone of the request to each middleware?

Reply:
was thinking the same - not sure what the overhead is or if there's any downside